### PR TITLE
Advance usage example

### LINKED ISSUE #1029: Documentation from Dogfooding Exercise
The main purpose of the document is something I encountered while dogfooding.  

I noticed that we're lacking examples of how to handle a more complex schema where you might be using a simple `object` and a polymorphic `object` with `templates`.

Luckily, I knew that we use a similar `schema` in the `tina-cloud-starter` and was able to unblock myself, but felt that we should have something in the documentation that goes over the process of writing the `schema`, writing the `query`, and writing the `render` - and how all of those parts fit together.

So far, I have a really rough draft attached to this issue (mostly just a paired down code example with some high level descriptions).

I think it could go with https://tina.io/docs/schema/ or https://tina.io/docs/graphql/ , but I'm honestly not sure.

[advanced-schema.md](https://github.com/tinacms/tinacms.org/files/7101767/advanced-schema.md)

### PR DESCRIPTION
**Pull request by perkinsjr**
Adding an advance usage that @Enigmatical created. 

Love some eyes on it. 

Fixes #1029

### CODE REVIEW

File: content/docs/schema-advanced.md [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+    }
+  }
+}
+```
+​
+## Rendering
+​
+The important step here is correctly retrieve the queried `data` out of `props`. The common pattern is `data: { query: { data: document }}`.
+​
+Notice that `sections` is iterated via `map()` utilizing the `__typename` to determine how each `section` should be rendered. You could also use `_template`.
```
Comment (jeffsee55) [inline content/docs/schema-advanced.md:236]:
```suggestion
Notice that `sections` is iterated via `map()` utilizing the `__typename` to determine how each `section` should be rendered.
```
`_template` isn't supported here, despite some misleading docs elsewhere (which should also be fixed)

File: content/docs/schema-advanced.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+      day: "numeric",
+      year: "numeric",
+    });
+  }
+​
+  return (
+    <div className="recipe-page">
+      {photo ? <img src={photo} alt="photo" />}
+      <h1>{name}</h1>
+      {localDatePublished ? <h2>{datePublished}</h2>}
```
Comment (jeffsee55) [inline content/docs/schema-advanced.md:271]:
```suggestion
      {localDatePublished ? <h2>{localDatePublished}</h2>}
```

File: content/docs/schema-advanced.md [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+  
+  let localDatePublished;
+  
+  if (datePublished) {
+    localDatePublished = new Date(datePublished).toLocaleDateString(undefined, {
+      month: "short",
+      day: "numeric",
+      year: "numeric",
+    });
+  }
```
Comment (jeffsee55) [inline content/docs/schema-advanced.md:265]:
```tsx
const localDatePublished = React.useMemo(() => {
  if (datePublished) {
    return new Date(datePublished).toLocaleDateString(undefined, {
      month: "short",
      day: "numeric",
      year: "numeric",
    });
  }
}, [datePublished)
```

**Review by jeffsee55 — COMMENTED** [review]
Just a couple of small ones

File: content/docs/schema-advanced.md [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+  
+  const localDatePublished = React.useMemo(() => {
+  if (datePublished) {
+    return new Date(datePublished).toLocaleDateString(undefined, {
+      month: "short",
+      day: "numeric",
+      year: "numeric",
+    });
+   }
+  }, [datePublished)
```
Comment (jeffsee55) [inline content/docs/schema-advanced.md:265]:
```suggestion
  }, [datePublished])
```
My bad

**Review by jeffsee55 — APPROVED** [review]

File: content/docs/schema-advanced.md (hunk 1 above)
Comment (Enigmatical) [reply content/docs/schema-advanced.md]: Thanks for catching this @jeffsee55 , I added that before I realized we weren't doing it.

**Pull request merged**